singularity/code/savegame.py

```diff
@@ -693,34 +693,35 @@ def savegame_exists(savegame_name):
 
     return True
 
-def sanitize_filename(filename):
-    """Create filename that's safe for all operating systems"""
+def check_filename_illegal(filename):
+    """Check if the filename is safe for all operating systems.
+
+    Returns an error message if a violation was found and None otherwise."""
+
+    result = None
 
     # https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx
     # http://www.linfo.org/file_name.html
     # https://kb.acronis.com/content/39790
 
     # Characters that are disallowed anywhere in a filename
     # No potential file separators or other potentially illegal characters
-    filename = re.sub('[<>:"|?*/\\\\]', '_', filename)
-
-    # Replace whitespace with _ and strip leading & trailing whitespace/_
-    filename = re.sub('\s', '_', filename).strip('_')
+    if re.match('.*[<>:"|?*/\\\\].*', filename):
+        return _('Do not use any of these characters in filename: {CHARACTERS}').format(CHARACTERS='<>:"|?*/\\\\')
```

Comment:
```suggestion
        return _('Filename must not contain any of these characters: {CHARACTERS}').format(CHARACTERS='<>:"|?*/\\\\')
```

(maybe with a line-break somewhere if the line is too long; it is a bit hard to tell in the github UI)

Reply:
Translations can potentially be longer than English anyway, and with a line break I'll need to add more vertical UI space. I didn't want to add more vertical space, because it will make other TextEntry dialogs that don't have a lot of text look crappy.

The alternative would be to inherit from the dialog or to change the size and positions from the outside, but that doesn't sound very appealing either.

I can make the UI wider if needed after I have translated it.